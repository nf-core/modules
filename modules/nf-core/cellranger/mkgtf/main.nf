process CELLRANGER_MKGTF {
    tag "$gtf"
    label 'process_low'

    container "nfcore/cellranger:7.0.0"

    // Exit if running this module with -profile conda / -profile mamba
    if (workflow.profile.tokenize(',').intersect(['conda', 'mamba']).size() >= 1) {
        exit 1, "CELLRANGER_MKGTF module does not support Conda. Please use Docker / Singularity / Podman instead."
    }

    input:
    path gtf

    output:
    path "*.filtered.gtf", emit: gtf
    path "versions.yml"  , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    """
    cellranger \\
        mkgtf \\
        $gtf \\
        ${gtf.baseName}.filtered.gtf \\
        $args

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        cellranger: \$(echo \$( cellranger --version 2>&1) | sed 's/^.*[^0-9]\\([0-9]*\\.[0-9]*\\.[0-9]*\\).*\$/\\1/' )
    END_VERSIONS
    """
}
