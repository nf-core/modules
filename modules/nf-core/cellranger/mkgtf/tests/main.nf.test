// nf-core modules test cellranger/mkgtf
nextflow_process {

    name "Test Process CELLRANGER_MKGTF"
    script "../main.nf"
    process "CELLRANGER_MKGTF"

    tag "modules"
    tag "modules_nfcore"
    tag "cellranger"
    tag "cellranger/mkgtf"

    test("homo_sapiens - reference") {

        when {
            process {
                """
                input[0] = file(params.test_data['homo_sapiens']['genome']['genome_gtf'], checkIfExists: true)
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("homo_sapiens - reference - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = file(params.test_data['homo_sapiens']['genome']['genome_gtf'], checkIfExists: true)
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
