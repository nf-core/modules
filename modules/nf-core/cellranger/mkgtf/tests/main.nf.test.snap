{
    "homo_sapiens - reference": {
        "content": [
            {
                "0": [
                    "genome.filtered.gtf:md5,50fc877b1c53b36b3b413aff88bda48c"
                ],
                "1": [
                    "versions.yml:md5,440fb677a34013b092e10c5daa9916e9"
                ],
                "gtf": [
                    "genome.filtered.gtf:md5,50fc877b1c53b36b3b413aff88bda48c"
                ],
                "versions": [
                    "versions.yml:md5,440fb677a34013b092e10c5daa9916e9"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-22T08:51:17.576906345"
    },
    "homo_sapiens - reference - stub": {
        "content": [
            {
                "0": [
                    "genome.filtered.gtf:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "1": [
                    "versions.yml:md5,440fb677a34013b092e10c5daa9916e9"
                ],
                "gtf": [
                    "genome.filtered.gtf:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "versions": [
                    "versions.yml:md5,440fb677a34013b092e10c5daa9916e9"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-22T08:51:23.23741543"
    }
}